sphinx/writers/latex.py

```diff
@@ -1378,19 +1360,25 @@ def depart_hlistcol(self, node):
         pass
 
     def latex_image_length(self, width_str):
-        match = re.match('(\d*\.?\d*)\s*(\S*)', width_str)
+        return self.latex_dim_to_latexdim(width_str)
+
+    def latex_dim_to_latexdim(self, length_str):
+        match = re.match('(\d*\.?\d*)\s*(\S*)', length_str)
         if not match:
             # fallback
-            return width_str
-        res = width_str
+            return length_str
         amount, unit = match.groups()[:2]
         if not unit:
-            return None
+            length_str = '%s\\sphinxpxdimen' % amount
+        elif unit == "pt":
+            length_str = '%sbp' % amount  # convert to 'bp'
+        # px unit not natively accepted by all engines, use \sphinxpxdimen
         elif unit == "px":
-            res = "%.3f\\sphinxpxdimen" % (float(amount))
+            length_str = "%s\\sphinxpxdimen" % amount
```

Comment:
I don't see why the ``.3f`` conversion should be needed. TeX accepts ``\d*\.?\d*`` in front of a dimension.